improver_tests/snow_fraction/test_SnowFraction.py

```diff
@@ -0,0 +1,161 @@
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Tests of SnowFraction plugin"""
+
+from datetime import datetime
+
+import iris
+import numpy as np
+import pytest
+from cf_units import Unit
+
+from improver.snow_fraction import SnowFraction
+from improver.synthetic_data.set_up_test_cubes import (
+    construct_scalar_time_coords,
+    set_up_variable_cube,
+)
+
+COMMON_ATTRS = {
+    "source": "Unit test",
+    "institution": "Met Office",
+    "title": "Post-Processed IMPROVER unit test",
+}
+RAIN_DATA = np.array([[0, 0.0], [0.5, 1]], dtype=np.float32)
+SNOW_DATA = np.array([[0, 0.5], [1, 0.0]], dtype=np.float32)
+
+
+def setup_cubes(rain_data=RAIN_DATA, snow_data=SNOW_DATA, name="{phase}rate"):
+    """Make CF-compliant rain and snow cubes from supplied arrays"""
+    if "rate" in name:
+        units = "m s-1"
+        time_bounds = None
+    else:
+        units = "m"
+        time_bounds = (datetime(2017, 11, 10, 3, 0), datetime(2017, 11, 10, 4, 0))
+    rain = set_up_variable_cube(
+        rain_data,
+        name.format(phase="rain"),
+        units,
+        time_bounds=time_bounds,
+        spatial_grid="equalarea",
+        attributes=COMMON_ATTRS,
+    )
+    snow = set_up_variable_cube(
+        snow_data,
+        name.format(phase="lwe_snow"),
+        units,
+        time_bounds=time_bounds,
+        spatial_grid="equalarea",
+        attributes=COMMON_ATTRS,
+    )
+    return rain, snow
+
+
+@pytest.mark.parametrize(
+    "cube_name", ("{phase}rate", "thickness_of_{phase}fall_amount")
+)
+def test_basic(cube_name):
+    """Run a test with four values, including one that will trigger divide-by-zero.
+    Check data and metadata of result."""
+    rain, snow = setup_cubes(name=cube_name)
+
+    expected_data = np.array([[1.0, 1.0], [2.0 / 3.0, 0.0]], dtype=np.float32)
+    result = SnowFraction()(iris.cube.CubeList([rain, snow]))
+    assert isinstance(result, iris.cube.Cube)
```

Comment:
I had a similar problem with codacy quality review in my field_texture unit tests because of this. If it's something that needs to be solved then it could work by splitting the data and metadata checks into 2 separate tests.

Reply:
I don't think Codacy is complaining about this - it thinks my code is too complex (which I disagree with and so did a couple of green team consultants)